### PR TITLE
Refactor Kyuubi kubernetes block cleaner

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

* fix bug as a blockMgr dir is created with no file at the begining.
* simplify code
* simplify log
* add test

### _How was this patch tested?_
the release directory structure
image

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @zwangsheng @yaooqinn

File: tools/spark-block-cleaner/src/main/scala/org/apache/kyuubi/tools/KubernetesSparkBlockCleaner.scala [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      info(s"start all clean job")
+      val startTime = System.currentTimeMillis()
+      val hasFinished = new CountDownLatch(cacheDirs.length)
+      cacheDirs.foreach { dir =>
+        threadPool.execute(() => {
+          doCleanJob(dir)
+          hasFinished.countDown()
+        })
+      }
+      hasFinished.await()
```
Comment (zwangsheng) [inline tools/spark-block-cleaner/src/main/scala/org/apache/kyuubi/tools/KubernetesSparkBlockCleaner.scala:186]: Here waiting for all the threads finsh jobs， means if usedtime greater than scheduleInterval, we can not start next cleaner on time.

**Review by yaooqinn — DISMISSED** [review]

File: tools/spark-block-cleaner/src/main/scala/org/apache/kyuubi/tools/KubernetesSparkBlockCleaner.scala (hunk 1 above)
Comment (ulysses-you) [reply tools/spark-block-cleaner/src/main/scala/org/apache/kyuubi/tools/KubernetesSparkBlockCleaner.scala:186]: yes, see the warn log. I think it'ok and it's no meaning if schedule interval is too small.

Comment (zwangsheng) [reply tools/spark-block-cleaner/src/main/scala/org/apache/kyuubi/tools/KubernetesSparkBlockCleaner.scala:186]: I got the idea if two thread cleaner one specific dir may trigger thread problem.LGTM.

**Review by zwangsheng — APPROVED** [review]

**Review by yaooqinn — DISMISSED** [review]

**Review by yaooqinn — APPROVED** [review]

**Review by zwangsheng — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master for v1.3.0

**Pull request merged**